src/commands/sgx.rs

```diff
@@ -0,0 +1,79 @@
+//! `tmkms sgxsign` CLI (sub)commands
+
+use crate::keyring::SecretKeyEncoding;
+use subtle_encoding::encoding::Encoding;
+use crate::prelude::*;
+use signatory_sgx::provider::SgxSigner;
+use std::path::PathBuf;
+use abscissa_core::{Command, Help, Runnable};
+
+/// The `softsign` subcommand
+#[derive(Command, Debug, Options, Runnable)]
+pub enum SgxCommand {
+    /// Show help for the `sgx` subcommand
+    #[options(help = "show help for the 'yubihsm' subcommand")]
+    Help(Help<Self>),
+
+    /// Generate a software signing key
+    #[options(help = "generate a sgx signing secret key")]
+    Keygen(KeygenCommand),
+
+    /// Generate a software signing key
+    #[options(help = "get the sgx signing public key")]
+    Pubkey(PubkeyCommand),
+}
+
+
+/// `keygen` command
+#[derive(Command, Debug, Default, Options)]
+pub struct KeygenCommand {
+    #[options(free, help = "file path where generated secret key should be created")]
+    secret_key_path: PathBuf,
+
+    #[options(
+        short = "s",
+        help = "sgx server address in Ip:port format",
+        default = "127.0.0.1:8888",
+    )]
+    sgx_server: String,
```

Comment:
so this assumes, someone runs this: https://github.com/crypto-com/signatory/commit/b35f79ebf8ad6476554fde575853312af8af34f5#diff-3749f882220cd92e3566832bf95b1ab8 ?